ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/TaskList/Decision/DecisionDate.cshtml.cs

```diff
@@ -25,21 +26,25 @@ public class DecisionDate : DecisionBaseModel
 		[Required(ErrorMessage = "Please enter a decision date"), Display(Name = "Decision")]
 		public DateTime? DateOfDecision { get; set; }
 
+		  public string DecisionText { get; set; }
+
 		public LinkItem GetPageForBackLink(int id)
 		{
 			var decision = GetDecisionFromSession(id);
 
-			return decision.ApprovedConditionsSet switch
+			return decision.Decision switch
 			{
-				true => Links.Decision.WhatConditions,
-				_ => Links.Decision.AnyConditions,
+				AdvisoryBoardDecisions.Approved => decision is { ApprovedConditionsSet : true } ? Links.Decision.WhatConditions : Links.Decision.AnyConditions,
```

Comment:
This syntax doesn't look quite right to me. 
`return decision switch
			{
				{ Decision: AdvisoryBoardDecisions.Approved, ApprovedConditionsSet: true } => Links.Decision.WhatConditions,
				{ Decision: AdvisoryBoardDecisions.Approved, ApprovedConditionsSet: false } => Links.Decision.AnyConditions,
				{ Decision: AdvisoryBoardDecisions.Declined } => Links.Decision.DeclineReason,
				_ => throw new Exception("Unexpected decision state")
			};`
Now I've typed it all out I'm not convinced it makes it more readable. 
@MartinWheelerMT  what do you think?

Reply:
This, and a few other places, is the bits I mentioned about extracting out into a navigation "thing" (attempting to avoid solutionizing early...) pass in the Project decision and current page (from Links) and it could be in charge of where 'Back' and 'Submit' point.